hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieColumnProjectionUtils.java

```diff
@@ -112,22 +112,21 @@ public static List<Pair<String,String>> getIOColumnNameAndTypes(Configuration co
   /**
    * If schema contains timestamp columns, this method is used for compatibility when there is no timestamp fields.
    *
-   * <p>We expect 3 cases to use parquet-avro reader {@link org.apache.hudi.hadoop.avro.HoodieAvroParquetReader} to read timestamp column:
+   * <p>We expect 2 cases to use parquet-avro reader {@link org.apache.hudi.hadoop.avro.HoodieAvroParquetReader} to read timestamp column:
    *
    * <ol>
    *   <li>Read columns contain timestamp type;</li>
    *   <li>Empty original columns;</li>
-   *   <li>Empty read columns but existing original columns contain timestamp type.</li>
    * </ol>
    */
   public static boolean supportTimestamp(Configuration conf) {
     List<String> readCols = Arrays.asList(getReadColumnNames(conf));
     if (readCols.isEmpty()) {
-      return getIOColumnTypes(conf).contains("timestamp");
```

Comment:
@xicm Here I think it should return false directly, what do you think.

Reply:
Agree with you, @cdmikechen do you have any other concern?

Reply:
When the `readCols` can be empty ?

Reply:
In such case, the timestamp can be read correctly anyway?

Reply:
> In such case, the timestamp can be read correctly anyway?

yes